### ISSUE TITLE
DisableAppSuggestions errors

### ISSUE THREAD

**Issue by r3incarnat0r** [Disassembler0/Win10-Initial-Setup-Script#148]
I've got this error while applying DisableAppSuggestions tweak. I just throwing the output here, didn't do any research regarding these errors, except trying to run once more without admin rights (same errors). Lemme know if you need to check or test something. Windows 10 Pro x64 1809.
```
Disabling Application suggestions...
Get-ItemProperty : Cannot bind argument to parameter 'Path' because it is null.
At C:\(MyPath)\Win10-Initial-Setup-Script\Win10.psm1:141 char:35
+         $data = (Get-ItemProperty -Path $key.PSPath -Name "Data").Dat ...
+                                         ~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Get-ItemProperty], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.GetItemProp
   ertyCommand

Set-ItemProperty : Cannot bind argument to parameter 'Path' because it is null.
At C:\(MyPath)\Win10-Initial-Setup-Script\Win10.psm1:142 char:26
+         Set-ItemProperty -Path $key.PSPath -Name "Data" -Type Binary  ...
+                                ~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Set-ItemProperty], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.SetItemProp
   ertyCommand
```

**Comment by Disassembler0** [comment]
That almost sounds like you've hit a moment when the cache got flushed and the registry tree is empty.

Can you please check if you have anything under `HKCU\SOFTWARE\Microsoft\Windows\CurrentVersion\CloudStore\Store\Cache\DefaultAccount`? Either there won't be anything or there will be a tree with bunch of subkeys with GUID in them.

If you do have the tree, check if you have there a subkey ending with `windows.data.placeholdertilecollection`. If you do have even that, try to rerun the tweak. If you don't log out, log back in and check it again.

Is this a clean system or already used? If used, did you previously applied the GPO-version of `DisableSharedExperiences`? (i.e. do you have `EnableCdp` and `EnableMmx` under `HKLM\SOFTWARE\Policies\Microsoft\Windows\System`? ) If so, the try to remove them and reapply the tweak using non-GPO variant as the GPO is now known to cause problems with start menu caches (#145).

**Comment by r3incarnat0r** [comment]
I do have *placeholdertilecollection key. Tried thw tweak again and still got errors. Windows is upgraded from previous build. I removed EnableCdp and EnableMmx, reaplied DisableSharedExperiences and then DisableAppSuggestions went fine without errors. I also had similar problem as #145 - I could not get rid of sections for boxes and Edge reappeared every restart. It seems to be ok now.

**Comment by r3incarnat0r** [comment]
Yesterday I reenabled gpo version of DisableSharedExperiences and was curious if I see the problems again, but after restart everything still went ok. Anyway the problems appeared again today after another new start. What I wanted to mention is that even though I get null errors from script, the Data value in registry is present.

**Comment by Disassembler0** [comment]
Apparently you have a quantum computer. Otherwise it's impossible to have the keys and not be able to access them at the same time. :) Anyway, I've pushed the commit with simplification of the registry operations. If you're gonna try again and still won't find anything, feel free to blame it on #145 and close this issue.

**Comment by r3incarnat0r** [comment]
I reproduced the problem...sort of. If I add/keep those GPOs to the registry, in some time (not after restart but more like next day) the content under CloudStore\Store vanishes. If I apply DisableAppSuggestions tweak, it's first run fails, but goes ok without error if I run it again, because the keys are instantly created after executing "Stop-Process -Name "ShellExperienceHost"..." - that's what the script calls purging cache I assume, but doesn't count with nonexisting keys and Data value. If I comment out this line in script, all of the four approaches above fail complaining about either null exception or nonexisting path as expected.

The original mystery still remains...why the errors was keeping displayed if I run the tweak repeatedly (and you added purging the cache together with reading the keys in commit 503b1a2) a few days ago? I suspect Microsoft changes GPOs/CloudStore behavior a little since KB4464330 as they're hopefully trying to fix the buggy 1809 build. And according to Windows Update history, I have KB4464330 installed since 09.10. (on the second attempt, as many other users do), which chronologically matches. I suppose (or at least hope) this weird thing (besides other things) will be fixed after 1809 re-release, so the GPOs could be safely applied again.

So yes, I ended up blaming Microsoft as you predicted in the previous comment and I'm not gonna waste any more time on this. Maybe except checking if the "final" 1809 will change something. :D
